pinot-spi/src/main/java/org/apache/pinot/spi/utils/retry/RetryPolicy.java

```diff
@@ -33,7 +33,8 @@ public interface RetryPolicy {
    * @param operation The operation to attempt, which returns true on success and false on failure.
    * @throws AttemptsExceededException
    * @throws RetriableOperationException
+   * @return the number of attempts used for the operation. 0 means the first try was successful.
    */
-  void attempt(Callable<Boolean> operation)
+  int attempt(Callable<Boolean> operation)
```

Comment:
can we add some tests in RetryPolicyTest for this new return value?

Reply:
The challenge is when you throw, you don't return a value. I could assume is the max Retry count and add that, else this change continues to sprawl by adding a new out arg to that method that may need to change everything that uses it.

Reply:
Done